token_verifier_jwt.go

```diff
@@ -0,0 +1,118 @@
+package centrifuge
+
+import (
+	"crypto/rsa"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+type tokenVerifierJWT struct {
+	mu                 sync.RWMutex
+	TokenHMACSecretKey []byte
+	TokenRSAPublicKey  *rsa.PublicKey
+}
+
+func newTokenVerifierJWT(tokenHMACSecretKey string, tokenRSAPublicKey *rsa.PublicKey) tokenVerifier {
+	return &tokenVerifierJWT{
+		TokenHMACSecretKey: []byte(tokenHMACSecretKey),
+		TokenRSAPublicKey:  tokenRSAPublicKey,
+	}
+}
+
+var (
+	errTokenExpired   = errors.New("token expired")
+	errMalformedToken = errors.New("malformed token")
+)
+
+func (verifier *tokenVerifierJWT) VerifyConnectToken(token string) (connectToken, error) {
+	parsedToken, err := jwt.ParseWithClaims(token, &connectTokenClaims{}, verifier.jwtKeyFunc())
+	if err != nil {
+		if err, ok := err.(*jwt.ValidationError); ok {
+			if err.Errors == jwt.ValidationErrorExpired {
+				// The only problem with token is its expiration - no other
+				// errors set in Errors bitfield.
+				return connectToken{}, errTokenExpired
+			}
+		}
+		return connectToken{}, err
+	}
+	if claims, ok := parsedToken.Claims.(*connectTokenClaims); ok && parsedToken.Valid {
+		token := connectToken{
+			UserID:   claims.StandardClaims.Subject,
+			ExpireAt: claims.StandardClaims.ExpiresAt,
+			Info:     claims.Info,
+		}
+		if claims.Base64Info != "" {
+			byteInfo, err := base64.StdEncoding.DecodeString(claims.Base64Info)
+			if err != nil {
+				return connectToken{}, err
+			}
```

Comment:
@FZambia you don't want disconnect the client in Base64 Info anymore ?

Reply:
I decided to simplify logic to only 2 cases: `token expired` case and all other `invalid token` case. In case of wrong base64 here error will result into `DisconnectInvalidToken` - i.e. client will still be disconnected.